modules/gdscript/doc_classes/@GDScript.xml

```diff
@@ -1363,6 +1363,26 @@
 				Stops the function execution and returns the current suspended state to the calling function.
 				From the caller, call [method GDScriptFunctionState.resume] on the state to resume execution. This invalidates the state. Within the resumed function, [code]yield()[/code] returns whatever was passed to the [code]resume()[/code] function call.
 				If passed an object and a signal, the execution is resumed when the object emits the given signal. In this case, [code]yield()[/code] returns the argument passed to [code]emit_signal()[/code] if the signal takes only one argument, or an array containing all the arguments passed to [code]emit_signal()[/code] if the signal takes multiple arguments.
+				You can also use [code]yield[/code] to wait for a function to finish:
+				[codeblock]
+				func _ready -> void:
+				    yield(do_something(), "completed")
+				    yield(do_something_else(), "completed")
+				    print("All functions are done!")
+
+				func do_something():
+				    print("Something is done!")
+				
+				func do_something_else():
+				    print("Something else is done!")
+				
+				# prints:
+				# Something is done!
+				# Something else is done!
+				# All functions are done!
+				[/codeblock]
+				The [code]completed[/code] signal is automatically created by the engine if the first argument is a function and will only be emitted once to resume [code]yield[/code]. For this reason it is necessary to include [code]completed[/code] as [code]signal[/code] when yielding with functions.
```

Comment:
It still reads a bit confusing to me. How about something like this, if it's accurate:
```
When yielding on a function, the [code]completed[/code] signal will be emitted automatically when the function returns. It can therefore be connected to as the target [code]signal[/code] for [code]yield[/code] to resume as soon as the function has been processed.
```

Reply:
Yeah, the concept itself is confusing. As @neikeq pointed in he's pull request (i cant find it >.<) that adds this functionality, the signal should be ditched altogether since it makes no sense returning it. So you could probably do something like `yield(do_something())` without the signal and yield will wait untill the function returns. But i guess they never worked on that.

> It can therefore be connected to as the target [code]signal[/code] for [code]yield[/code] to resume as soon as the function has been processed.

That part is still confusing. How about:

> It can, therefore, be used as the [code]signal[/code] parameter of [code]yield[/code] method without the need of creating a signal first.

Reply:
> It can, therefore, be used as the [code]signal[/code] parameter of [code]yield[/code] method to resume, without the need of creating a signal first.

Reply:
I don't think you need to talk about creating or emitting a signal. We already say that the engine emits it.

> When yielding on a function, a [code]completed[/code] signal will be emitted automatically when the function returns. You can therefore use [code]"completed"[/code] as the target [code]signal[/code] of the [code]yield[/code] call.

Reply:
So:
> It can, therefore, be used as the [code]signal[/code] parameter of [code]yield[/code] method to resume.

Reply:
If you *really* want the insert, sure. But there should be`the` before `[code]yield[/code]`, you can't say "of method" in English.